Gems/Atom/Tools/MaterialCanvas/Code/Source/MaterialCanvasApplication.cpp

```diff
@@ -77,14 +81,15 @@ namespace MaterialCanvas
 
         // Register all data types required by material canvas nodes with the dynamic node manager
         m_dynamicNodeManager->RegisterDataTypes({
-            AZStd::make_shared<GraphModel::DataType>(MaterialCanvasDataTypeEnum::Int32, int32_t{}, "int32"),
-            AZStd::make_shared<GraphModel::DataType>(MaterialCanvasDataTypeEnum::Uint32, uint32_t{}, "uint32"),
-            AZStd::make_shared<GraphModel::DataType>(MaterialCanvasDataTypeEnum::Float32, float{}, "float"),
-            AZStd::make_shared<GraphModel::DataType>(MaterialCanvasDataTypeEnum::Vector2, AZ::Vector2::CreateZero(), "Vector2"),
-            AZStd::make_shared<GraphModel::DataType>(MaterialCanvasDataTypeEnum::Vector3, AZ::Vector3::CreateZero(), "Vector3"),
-            AZStd::make_shared<GraphModel::DataType>(MaterialCanvasDataTypeEnum::Vector4, AZ::Vector4::CreateZero(), "Vector4"),
-            AZStd::make_shared<GraphModel::DataType>(MaterialCanvasDataTypeEnum::Color, AZ::Color::CreateOne(), "Color"),
-            AZStd::make_shared<GraphModel::DataType>(MaterialCanvasDataTypeEnum::String, AZStd::string{}, "String"),
+            AZStd::make_shared<GraphModel::DataType>(AZ_CRC_CE("bool"), bool{}, "bool"),
```

Comment:
There might be an opportunity to use the AZ::Name class to improve GraphModel overall, including enum types. (GraphModel was built before the Name class and ever since then I've been interested in updating GraphModel to use it).

Reply:
Another option might be to use the AZ::TypeId for these types rather than a CRC of the name.

Reply:
@santorac I was thinking the same thing. I spoke with @cgalvan about getting rid of the enum so we're not defining the same set of types in multiple places. Enum is used all over landscape canvas but can be phased out in another change. 

There are currently multiple ways to look up the data types: display name, CPP name, type ID, enum. Display name and enum can be combined using AZ Name. Fortunately, it seems like none of this is serialized so making these kinds of changes won't impact anyone else.

Landscape canvas is registering the same data type multiple times with custom type IDs as a way to force support for context sensitive connections between entities. We discussed other ways to approach this like subclassing the underlying data type with a custom type ID. However, being able to bind the same underlying type with different unique type IDs could be useful to distinguish between numeric or other data types that share the same underlying type.

Beyond that, the CPP name and actual type ID can be obtained directly from the default value, which I implemented with other template constructors.